cpp/src/Ice/Network.cpp

```diff
@@ -2174,7 +2175,7 @@ IceInternal::getHostName()
     char name[256];
     if (gethostname(name, sizeof(name)) != 0)
     {
-        throw SocketException(__FILE__, __LINE__, getSocketErrno());
+        throw SocketException{__FILE__, __LINE__, getSocketErrno()};
```

Comment:
The `(...)` constructor syntax worked despite the int / unsigned long mismatch on Windows. But not the `{...}` syntax. I don't understand why.

Reply:
With the initializer list narrowing conversions are disallowed.